### PR TITLE
[ML] add supported types to no fields error message

### PR DESCRIPTION
**Pull request by benwtrent**
Minor change, users may want to know what a "supported field" is. It is simple enough to indicate the supported field types in the error message.

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java [hunk 1]
```diff
@@ -84,7 +84,13 @@ public ExtractedFields detect() {
         checkRequiredFieldsArePresent(fields);
 
         if (fields.isEmpty()) {
-            throw ExceptionsHelper.badRequestException("No compatible fields could be detected in index {}", Arrays.toString(index));
+            Set<String> supportedTypes = new HashSet<>(NUMERICAL_TYPES);
+            if (config.getAnalysis().supportsCategoricalFields()) {
+                supportedTypes.addAll(CATEGORICAL_TYPES);
+            }
+            throw ExceptionsHelper.badRequestException("No compatible fields could be detected in index {}. Supported types are {}.",
```
Comment (dimitris-athanasiou) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java]: Nice! We should also do the same for the last error message in `removeFieldsWithIncompatibleTypes()`

Comment (benwtrent) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java]: roger

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by dimitris-athanasiou** [comment]
run elasticsearch-ci/1

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

**Pull request merged**